api/src/main/java/org/commonjava/maven/galley/io/SpecialPathConstants.java

```diff
@@ -56,20 +52,36 @@
 
         sp.add( pi );
 
-        pi = SpecialPathInfo.from( new FilePatternMatcher( "maven-metadata\\.xml(\\.md5|\\.sha[\\d]+)?$" ) )
+        pi = SpecialPathInfo.from( new FilePatternMatcher( "maven-metadata\\.xml$" ) )
                             .setMergable( true )
                             .setMetadata( true )
                             .build();
 
         sp.add( pi );
 
-        pi = SpecialPathInfo.from( new FilePatternMatcher( "archetype-catalog\\.xml(\\.md5|\\.sha[\\d]+)?$" ) )
+        pi = SpecialPathInfo.from( new FilePatternMatcher( "maven-metadata\\.xml(\\.md5|\\.sha[\\d]+)$" ) )
+                .setDecoratable( false )
+                .setMergable( true )
+                .setMetadata( true )
+                .build();
```

Comment:
You are right, but in this case the meaning of the flag is "in case of group don't pick the first matching file from a constituent but rather use a generator to produce a result from all constituent, not necessarily from files with the same name". Maybe the name of the flag should be different, but can't find anything short and fitting to match the above meaning :-) In any case I believe the value should be true to work properly.